src/main/java/manageezpz/logic/commands/EditTaskCommand.java

```diff
@@ -118,20 +122,73 @@ public CommandResult execute(Model model) throws CommandException {
         }
     }
 
+    /**
+     * If the value after a prefix is empty, return false.
+     * Else, return true.
+     */
+
+    boolean ensureFormatCompliance(HashMap<String, Boolean> prefixStatusHash,
+                                   String desc, String date, String time) {
+        boolean isFormatOkay = true;
+        HashMap<String, String> inputStatusHash = new HashMap<>();
+        inputStatusHash.put("description", desc);
+        inputStatusHash.put("date", date);
+        inputStatusHash.put("datetime", time);
+
+        String[] statusArr = {"description", "date", "datetime"};
+
+        for (String s : statusArr) {
+            boolean status = prefixStatusHash.get(s);
+            String input = inputStatusHash.get(s).trim();
+            if (status && input.isEmpty()) {
+                isFormatOkay = false;
+                break;
+            }
+        }
+        return isFormatOkay;
+    }
+
+    /**
+     * Updates a Todo task.
+     *
+     * @param currentTask Current Todo task that is to be updated
+     * @param desc New description of the Todo Task
+     * @return Updated Todo task
+     */
+
     private Task updateTodo(Todo currentTask, String desc) throws ParseException {
         Todo updatedToDoTask = new Todo(currentTask);
+        if (prefixStatusHash.get("date") || prefixStatusHash.get("datetime")) {
+            throw new ParseException(MESSAGE_EDIT_TODO_TASK_NO_DATE_AND_TIME_VALUES);
+        }
 
         if (!desc.isEmpty()) {
             Description newDesc = ParserUtil.parseDescription(desc);
             updatedToDoTask.setDescription(newDesc);
+        } else {
+            throw new ParseException(MESSAGE_EMPTY_DESCRIPTION);
```

Comment:
This is redundant because `ParserUtil.parseDescription` in line 166 would have already settled that. Hence the else condition will not be reached.